helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -626,6 +631,22 @@ private void handleEvent(ClusterEvent event, BaseControllerDataProvider dataProv
       return;
     }
 
+    // Event handling happens in a different thread from the onControllerChange processing thread.
+    // Thus, there are several possible conditions.
+    // 1. Event handled after leadership acquired. So we will have a valid rebalancer for the
+    // event processing.
+    // 2. Event handled shortly after leadership relinquished. And the rebalancer has not been
```

Comment:
Do you mean that the old controller will continue to process events even after having lost the leadership using the in-memory rebalancer/cache state?

Reply:
The one that is being processed will continue.